Fix link to list of supported integrations (#7804)
DataDog/documentation: content/en/tracing/setup/go.md

## Automatic Instrumentation

> Datadog has a series of pluggable packages which provide out-of-the-box support for instrumenting a series of libraries and frameworks. The list of supported integrations can be found in our [compatibility section](https://docs.datadoghq.com/tracing/compatibility_requirements/go/) (note that packages must be imported).

Reviewer: ```suggestion
Datadog has a series of pluggable packages which provide out-of-the-box support for instrumenting a series of libraries and frameworks. The list of supported integrations is in the [compatibility section](https://docs.datadoghq.com/tracing/compatibility_requirements/go/). Note that packages must be imported.
```
